### PR TITLE
Fix several issues when setting image and readme for spaces

### PR DESCRIPTION
**Pull request by JammingBen**
## Description
* An issue where setting a space-image or -readme would corrupt the file list has been fixed.
* Setting images from within the `.space` folder has been fixed.
* Setting images and readme files with specials characters in their names has been fixed.

## Related Issue

- Fixes https://github.com/owncloud/web/issues/6875

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>oCISSharingPublic1</strong> https://drone.owncloud.com/owncloud/web/25390/68/1
<details><summary>:boom: The acceptance tests failed on retry. Please find the screenshots inside ...</summary>

<p>


### webUISharingPublicBasic-publicLinkPublicActions_feature-L44.png
  


</p></details>
:boom: The acceptance tests pipeline failed. The build has been cancelled.

File: packages/web-app-files/src/mixins/spaces/actions/setReadme.js [hunk 1]
```diff
@@ -48,26 +47,26 @@ export default {
...
+        const storageId = this.$route.params.storageId
+        const space = this.spaces.find((s) => s.id === storageId)
+
         const fileContent = await this.$client.files.getFileContents(resources[0].webDavPath)
         const fileMetaData = await this.$client.files.putFileContents(
           `/spaces/${space.id}/.space/readme.md`,
           fileContent
         )
-        this.UPDATE_RESOURCE_FIELD({
+        this.UPDATE_SPACE_FIELD({
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/spaces/actions/setReadme.js:62]: I think you need to do both here. The space can also be the current folder if you're setting the readme from within the project root folder.

File: packages/web-app-files/src/mixins/spaces/actions/setReadme.js [hunk 2]
```diff
@@ -48,26 +47,26 @@ export default {
...
   methods: {
-    ...mapMutations('Files', ['UPDATE_RESOURCE_FIELD']),
+    ...mapMutations('Files', ['UPDATE_SPACE_FIELD']),
     ...mapActions(['showMessage']),
     async $_setSpaceReadme_trigger({ resources }) {
-      const space = this.currentFolder
-
       try {
+        const storageId = this.$route.params.storageId
+        const space = this.spaces.find((s) => s.id === storageId)
```
Comment (kulmann) [inline packages/web-app-files/src/mixins/spaces/actions/setReadme.js:55]: The `space` was set in the folder loader before, see `loaderProject.ts` line 77. Can/should we remove that from the folder loader then?

File: packages/web-app-files/src/mixins/spaces/actions/setReadme.js (hunk 1 above)
Comment (JammingBen) [reply packages/web-app-files/src/mixins/spaces/actions/setReadme.js:62]: This should be sufficient as far as I can see. The README data gets fetched from the space in `state.spaces`, which gets updated via `UPDATE_SPACE_FIELD `.

Comment (kulmann) [reply packages/web-app-files/src/mixins/spaces/actions/setReadme.js:59]: ok 👍

File: packages/web-app-files/src/services/folder/spaces/loaderProject.ts [hunk 3]
```diff
@@ -74,6 +74,7 @@ export class FolderLoaderSpacesProject implements FolderLoader {
       })
 
       if (!sameRoute) {
+        ref.UPSERT_SPACE(space)
```
Comment (kulmann) [inline packages/web-app-files/src/services/folder/spaces/loaderProject.ts:77]: Doesn't hurt to upsert the space outside of this `if` statement. Even if it's not the displayed space, we'd stell get updated data in the store without extra cost.

**Pull request merged**